### PR TITLE
Use config to determine if UA should be omitted

### PR DESCRIPTION
**Pull request by SlayterDev**
Task/Issue URL: https://app.asana.com/0/1200890834746050/1200932417493197/f
Tech Design URL:
CC: @brindy 

**Description**:
This PR uses the config to see if the custom UserAgent should be omitted

**Tagging instructions**
If this PR isn't ready to be merged for whatever reason it should be marked with the `DO NOT MERGE` label (particularly if it's a draft)
If it's pending Product Review/PFR, please add the `Pending Product Review` label.

If at any point it isn't actively being worked on/ready for review/otherwise moving forward (besides the above PR/PFR exception) strongly consider closing it (or not opening it in the first place). If you decide not to close it, make sure it's labelled to make it clear the PRs state and comment with more information.

**Steps to test this PR**:
1. See https://github.com/duckduckgo/BrowserServicesKit/pull/95

**Device Testing**:

* [x] iPhone SE (1st Gen)
* [x] iPhone 8
* [x] iPhone X
* [x] iPad

**OS Testing**:

* [x] iOS 13
* [x] iOS 14
* [x] iOS 15

---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)
**When ready for review, remember to post the PR in MM**

### CODE REVIEW

**Review by brindy — APPROVED** [review]
LGTM! Feel free to merge once the config and BSK changes have been merged.

**Review by brindy — CHANGES_REQUESTED** [review]
Actually - there are test failures that look related. :)

**Review by brindy — APPROVED** [review]
Nice work. You have conflicts now 🙃  - but feel free to merge once they're resolved :)

File: DuckDuckGo.xcodeproj/project.xcworkspace/xcshareddata/swiftpm/Package.resolved
Comment (SlayterDev) [inline DuckDuckGo.xcodeproj/project.xcworkspace/xcshareddata/swiftpm/Package.resolved:37]: This was unintentional. Let me know and I can revert if needed.

**Review by brindy — APPROVED** [review]
LGTM, get it in :)

**Pull request merged**
